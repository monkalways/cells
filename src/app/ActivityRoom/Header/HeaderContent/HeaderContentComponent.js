import React from 'react';
import PropTypes from 'prop-types';
import { Avatar, Grid, withStyles } from '@material-ui/core';

import BailHearing1Icon from '../../../images/BailHearing1.png';
import BailHearing2Icon from '../../../images/BailHearing2.png';
import FingerprintingIcon from '../../../images/Fingerprinting.png';
import MedicalVisitIcon from '../../../images/MedicalVisit.png';
import PhoneAcceptIcon from '../../../images/PhoneAccept.png';
import InterviewIcon from '../../../images/Interview.png';
import BreathalyzerIcon from '../../../images/Breathalyzer.png';
import ReleaseRoomIcon from '../../../images/ReleaseHolding.png';
import RemandHoldingRoomIcon from '../../../images/RemandHolding.png';

const LOCATION_PHONE = 'phone';
const LOCATION_MEDICAL_VISIT = 'medical';
const LOCATION_FINGER_PRINTING = 'fingerprinting';
const LOCATION_BAIL_HEARING_1 = 'bail hearing1';
const LOCATION_BAIL_HEARING_2 = 'bail hearing2';
const LOCATION_INTERVIEW = 'interview';
const LOCATION_BREATH_TEST = 'breath test';
const LOCATION_RELEASE_ROOM = 'release room';
const LOCATION_REMAND_HOLDING_ROOM = 'remand holding room';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  usage: PropTypes.string.isRequired,
};

export const HeaderContentComponent = ({ classes, usage }) => {
  const getActivityRoomIcon = () => {
    let locationIconSrc = null;
    switch (usage.toLowerCase()) {
      case LOCATION_PHONE:
        locationIconSrc = PhoneAcceptIcon;
        break;
      case LOCATION_MEDICAL_VISIT:
        locationIconSrc = MedicalVisitIcon;
        break;
      case LOCATION_FINGER_PRINTING:
        locationIconSrc = FingerprintingIcon;
        break;
      case LOCATION_BAIL_HEARING_1:
        locationIconSrc = BailHearing1Icon;
        break;
      case LOCATION_BAIL_HEARING_2:
        locationIconSrc = BailHearing2Icon;
        break;
      case LOCATION_INTERVIEW:
        locationIconSrc = InterviewIcon;
        break;
      case LOCATION_BREATH_TEST:
        locationIconSrc = BreathalyzerIcon;
        break;
      case LOCATION_RELEASE_ROOM:
        locationIconSrc = ReleaseRoomIcon;
        break;
      case LOCATION_REMAND_HOLDING_ROOM:
        locationIconSrc = RemandHoldingRoomIcon;
        break;
      default:
        break;
    }
    return locationIconSrc;
  };

  return (
    <Grid container justify="center" className={classes.grid}>
      <Grid container alignItems="center" justify="center">
        <Avatar className={classes.avatar} src={getActivityRoomIcon()} />
      </Grid>
    </Grid>
  );
};

HeaderContentComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  grid: {
    width: theme.spacing.unit * 60,
  },
  avatar: {
    margin: theme.spacing.unit,
    height: theme.spacing.unit * 8,
    width: theme.spacing.unit * 8,
  },
}))(HeaderContentComponent);
