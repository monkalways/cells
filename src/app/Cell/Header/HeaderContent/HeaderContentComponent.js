import React from 'react';
import PropTypes from 'prop-types';
import { Grid, Typography, withStyles } from '@material-ui/core';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  cellDetails: PropTypes.shape({
    name: PropTypes.string.isRequired,
    designation: PropTypes.string.isRequired,
    occupancy: PropTypes.number.isRequired,
    occupancyCount: PropTypes.number.isRequired,
    cellStatus: PropTypes.string.isRequired,
  }).isRequired,
};

export const HeaderContentComponent = ({ classes, cellDetails }) => (
  <Grid container justify="center" className={classes.grid}>
    <Grid container alignItems="center" justify="center">
      <Grid item sm={5}>
        <Typography variant="h6" className={classes.headerTitle}>
          Cell:
        </Typography>
        <Typography variant="h4" className={classes.headerContent}>
          {cellDetails.name}
        </Typography>
      </Grid>
      <Grid item sm={7}>
        <Typography variant="h6" className={classes.headerTitle}>
          Designation:
        </Typography>
        <Typography variant="h6" className={classes.headerContent}>
          {cellDetails.designation}
        </Typography>
      </Grid>
      <Grid item sm={5}>
        <Typography variant="h6" className={classes.headerTitle}>
          Occupancy:
        </Typography>
        <Typography variant="h6" className={classes.headerContent}>
          {cellDetails.occupancyCount}/{cellDetails.occupancy}
        </Typography>
      </Grid>
      <Grid item sm={7}>
        <Typography variant="h6" className={classes.headerTitle}>
          Cell Status:
        </Typography>
        <Typography variant="h6" className={classes.headerContent}>
          {cellDetails.cellStatus}
        </Typography>
      </Grid>
    </Grid>
  </Grid>
);

HeaderContentComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  grid: {
    flexGrow: 1,
  },
  headerTitle: {
    display: 'inline',
    color: 'red',
    marginRight: theme.spacing.unit,
  },
  headerContent: {
    display: 'inline',
  },
}))(HeaderContentComponent);
