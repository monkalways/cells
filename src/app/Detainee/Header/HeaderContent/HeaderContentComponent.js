import React from 'react';
import PropTypes from 'prop-types';
import { Grid, Typography, withStyles } from '@material-ui/core';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  detainee: PropTypes.shape({
    assignedCellId: PropTypes.string.isRequired,
  }).isRequired,
};

const HeaderContent = ({ classes, detainee }) => (
  <Grid container justify="center" className={classes.grid}>
    <Grid item sm={5}>
      <Typography variant="h4" className={classes.headerTitle}>
        Cell:
      </Typography>
      <Typography variant="h4" className={classes.headerContent}>
        {detainee.assignedCellId.substring(0, 10)}
      </Typography>
    </Grid>
  </Grid>
);

HeaderContent.propTypes = propTypes;

export default withStyles((theme) => ({
  headerTitle: {
    display: 'inline',
    color: 'red',
    marginRight: theme.spacing.unit,
  },
  headerContent: {
    display: 'inline',
  },
}))(HeaderContent);
