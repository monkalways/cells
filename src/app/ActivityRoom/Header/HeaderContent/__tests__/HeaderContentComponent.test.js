import React from 'react';
import { shallow } from 'enzyme';
import { Avatar } from '@material-ui/core';

import { HeaderContentComponent } from '../HeaderContentComponent';
import BailHearing1Icon from '../../../../images/BailHearing1.png';
import BailHearing2Icon from '../../../../images/BailHearing2.png';
import FingerprintingIcon from '../../../../images/Fingerprinting.png';
import MedicalVisitIcon from '../../../../images/MedicalVisit.png';
import PhoneAcceptIcon from '../../../../images/PhoneAccept.png';
import InterviewIcon from '../../../../images/Interview.png';
import BreathalyzerIcon from '../../../../images/Breathalyzer.png';
import ReleaseRoomIcon from '../../../../images/ReleaseHolding.png';
import RemandHoldingRoomIcon from '../../../../images/RemandHolding.png';

describe('HeaderContentComponent', () => {
  let usage;

  beforeEach(() => {
    usage = 'phone';
  });

  const setup = () => {
    const classes = { grid: {}, avatar: {} };
    return shallow(<HeaderContentComponent classes={classes} usage={usage} />);
  };

  it('should render', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it.each([
    ['phone', PhoneAcceptIcon],
    ['bail hearing1', BailHearing1Icon],
    ['bail hearing2', BailHearing2Icon],
    ['medical', MedicalVisitIcon],
    ['fingerprinting', FingerprintingIcon],
    ['interview', InterviewIcon],
    ['breath test', BreathalyzerIcon],
    ['release room', ReleaseRoomIcon],
    ['remand holding room', RemandHoldingRoomIcon],
  ])(
    'should render avatar based on activity room usage',
    (activityRoomUsage, avatarSrc) => {
      usage = activityRoomUsage;
      const wrapper = setup();

      expect(wrapper.find(Avatar)).toHaveProp('src', avatarSrc);
    },
  );
});
