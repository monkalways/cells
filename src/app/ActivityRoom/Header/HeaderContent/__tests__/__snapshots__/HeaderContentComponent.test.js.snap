// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContentComponent should render 1`] = `
<WithStyles(Grid)
  className=""
  container={true}
  justify="center"
>
  <WithStyles(Grid)
    alignItems="center"
    container={true}
    justify="center"
  >
    <WithStyles(Avatar)
      className=""
      src="PhoneAccept.png"
    />
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
